components/org.wso2.carbon.identity.idp.metadata.saml2/src/main/java/org/wso2/carbon/identity/idp/metadata/saml2/util/SAML2SSOFederatedAuthenticatorConfigBuilder.java

```diff
@@ -62,260 +80,271 @@ private static FederatedAuthenticatorConfig parse(EntityDescriptor entityDescrip
 
         if (entityDescriptor != null) {
             List<RoleDescriptor> roleDescriptors = entityDescriptor.getRoleDescriptors();
-            //assuming only one IDPSSO is inside the entitydescripter
+            // Assuming that only one IDPSSODescriptor is inside the EntityDescriptor.
             if (CollectionUtils.isNotEmpty(roleDescriptors)) {
-                RoleDescriptor roleDescriptor = roleDescriptors.get(0);
-                if (roleDescriptor != null) {
-                    IDPSSODescriptor idpssoDescriptor;
-                    try {
-                        idpssoDescriptor = (IDPSSODescriptor) roleDescriptor;
-                    } catch (ClassCastException ex) {
-                        throw new IdentityApplicationManagementException("No IDP Descriptors found, invalid file content", ex);
+                IDPSSODescriptor idpssoDescriptor = null;
+                if (roleDescriptors.size() > 1) {
```

Comment:
This size check seems to be not required. Even if the size is 1, the for loop with do the job.

Reply:
Fixed with d108f743104046e4c5f08f17a0a6339982a7dea3.